pkg/build/builder/cmd/scmauth/password.go

```diff
@@ -86,7 +87,12 @@ func (u UsernamePassword) Setup(baseDir string) error {
 	fmt.Fprintf(gitconfig, UserPassGitConfig, gitcredentials.Name())
 	fmt.Fprintf(gitcredentials, "%s", usernamePasswordURL.String())
 
-	return ensureGitConfigIncludes(gitconfig.Name())
+	sourceURL := u.SourceURL
+	sourceURL.User = nil
```

Comment:
From here https://git-scm.com/docs/git-credential-store (at the end) it looks like git will try to match the username if one is provided in the URL. However if we strip the username and password from the URL, it should match on protocol and hostname which is fine, since we're only putting a single entry there. Because it will show in the logs, I'd prefer leaving the name out especially if you consider it a secret.